### DISCUSSION TITLE
Creating New Body changes Old Body to same position as New Body

### DISCUSSION THREAD

**Discussion by N8python** [enable3d/enable3d#121] [Help / Questions] [answered]
So, for my game, I create a physics body for an enemy using the following code:
```js
instance.third.add.existing(instance.enemy);
instance.third.physics.add.existing(instance.enemy, { shape: 'box', ignoreScale: true, offset: { y: -0.5 } });
```
(`instance` refers to the scene, `instance.enemy` is the FBX model in an `ExtendedObject3D()`)
Then, I try to add a new physics body, like this:
```js
instance.third.physics.add.box({ z: -3, y: 10 }, { lambert: { color: 'hotpink' } });
```
When this happens, the bounding box of the enemy (or `enemy.body`) suddenly becomes set to the box's body. The enemy itself doesn't teleport or anything, but when I try to access the enemies `body.position` and `body.rotation`, they are equal to that of the box.

Why would this be happening?

**Comment by yandeu** [comment]
Do you use the latest version? `@enable3d/phaser-extension@0.22.0-dev.1`

Looks to me like a bug. Can you share a simplified version of your project?

**Reply by N8python** [reply]
I will try and download that. If not I will share a simplified version.

**Reply by N8python** [reply]
Where can I find the @enable3d/phaser-extension@0.22.0-dev.1 version? I don't use npm - I just copy the `lib` directory and include the necessary scripts. (nvm found the file).

**Reply by N8python** [reply]
This also appears to happen with more than just the enemy - in fact, *every* boxes `body` attribute returns the body of the most frequent addition to the scene.

**Reply by N8python** [reply]
Updating to latest version did not fix problem.

**Comment by yandeu** [comment, marked as answer]
The properties `object.body.position` and `object.body.rotation` are reserved for headless mode only.

```ts
// physicsBody.ts

/** Get position. (for headless mode only) */
(property) PhysicsBody.position: {  x: number;  y: number;  z: number; }

/** Get the rotation in radians. (for headless mode only) */
(property) PhysicsBody.rotation: {  x: number;  y: number;  z: number; }
```
You should use `object.position` and  `object.rotation` instead.

---

If you really have to use `object.body.position` and `object.body.rotation`, you have to call `object.body.transform()` first.

```ts
// physicsBody.ts

/** You have to call transform() before you can get or set the body's position or rotation. (for headless mode only) */
(method) PhysicsBody.transform(): void

````

**Reply by N8python** [reply]
I will try that.

**Reply by N8python** [reply]
THANK YOU SO MUCH! A simple call to `.transform()` fixed the bug!
I appreciate how willing you are to help me with all my noob questions.
